src/zampy/recipe.py

```diff
@@ -123,12 +123,27 @@ def run(self) -> None:
 
             comp = dict(zlib=True, complevel=5)
             encoding = {var: comp for var in ds.data_vars}
-            fname = (  # e.g. "era5_2010-2020.nc"
-                f"{dataset_name.lower()}_{self.start_year}-{self.end_year}.nc"
-            )
+            time_start = str(self.timebounds.start.astype("datetime64[Y]"))
+            time_end = str(self.timebounds.end.astype("datetime64[Y]"))
+            # e.g. "era5_2010-2020.nc"
+            fname = f"{dataset_name.lower()}_{time_start}-{time_end}.nc"
             ds.to_netcdf(path=self.data_dir / fname, encoding=encoding)
 
         print(
             "Finished running the recipe. Output data can be found at:\n"
             f"    {self.data_dir}"
         )
+
+
+def convert_time(time: str) -> np.datetime64:
+    """Check input time and convert to np.datetime64."""
+    try:
+        timestamp = np.datetime64(time)
+    except ValueError as err:
```

Comment:
```suggestion
    except ValueError(match="enter text from numpy error here") as err:
```
If you add (part of) the numpy error here, you're sure to catch the right error and not miscatch a different one